xgcm/grid.py

```diff
@@ -1908,6 +1952,12 @@ def _apply_vector_function(self, function, vector, **kwargs):
         # the keys, should be axis names
         assert len(vector) == 2
 
+        warnings.warn(
+            "`interp_2d_vector` and `diff_2d_vector` will be removed from future releases."
+            "The same functionality will be available under the `xgcm.Grid` methods.",
+            category=DeprecationWarning,
```

Comment:
We could hopefully update this with a recommendation once we decide on what the vector logic API will look like?

Reply:
Yes, but I think that will depend on #410, no? Do you suggest to add a note in the code?

Reply:
I suppose it's like a chicken and egg problem.

Reply:
Lets just go with this now, and release today, so that we get the max warning time? Does that sound ok for you?